WeekFormulas.ts

```diff
@@ -203,27 +207,42 @@ function protectWeek(weekNum: number): void {
         wplProtection = getColumn(weekNum, 8).protect().setDescription("WPL PROTECTION");
     const sheetEditors = wahfProtection.getEditors();
 
-    wahfProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
+    //grab row length
+    
+    const proctectedSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName("Protection");
+    const calculateSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName("");
+    //grab row length
+
+    var range = getFormulaResult("=COUNTA(Protection!B1:1)");
+    const procDesk = proctectedSheet.getRange(1, 3, 1, range-1).getValues() as string[][];
+
+    range = getFormulaResult("=COUNTA(Protection!B2:2)");
+    const procStudy = proctectedSheet.getRange(2,  3, 1, range-1).getValues() as string[][];
+
+    range = getFormulaResult("=COUNTA(Protection!B3:3)");
+    const procSeminar = proctectedSheet.getRange(3, 3, 1, range-1).getValues() as string[][];
+
+    range = getFormulaResult("=COUNTA(Protection!B4:4)");
+    const procDbase = proctectedSheet.getRange(4, 3, 1, range-1).getValues() as string[][];
+
     semProtection
         .removeEditors(sheetEditors)
-        .addEditors(
-            databaseAdmin.concat(seminarFiller.tls.map((tl) => tl.email))
-        );
+        .addEditors(procSeminar[0]);
     ssProtection
         .removeEditors(sheetEditors)
-        .addEditors(
-            databaseAdmin.concat(studySessionFiller.tls.map((tl) => tl.email))
-        );
+        .addEditors(procStudy[0]);
     fdProtection
         .removeEditors(sheetEditors)
-        .addEditors(
-            databaseAdmin.concat(frontDeskFiller.tls.map((tl) => tl.email))
-        );
-    zoomProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
-    tlmProtection.removeEditors(sheetEditors).addEditors(databaseAdmin);
+        .addEditors(procDesk[0]);
+
+
+    wahfProtection.removeEditors(sheetEditors).addEditors(procDbase[0]);
```

Comment:
For every single protection, all database admin need to be added as well. But the way this should be done is similar to your implementation here; a row in the Protection sheet should be dedicated to the emails of the database admin. Then on each `addEditors()` call, you would also pass in these database admins like so:
`fdProtection.addEditors([...databaseAdmin, ...procDesk])`